tempesta_fw/http_msg.c

```diff
@@ -311,6 +372,104 @@ __hdr_lookup(TfwHttpMsg *hm, const TfwStr *hdr)
 	return id;
 }
 
+/**
+ * Special procedure comparing specified name against the header in HTTP/2
+ * or HTTP/1.1 format.
+ */
+int
+__hdr_name_cmp(const TfwStr *hdr, const TfwStr *name)
+{
+	long n;
+	int i1, i2, off1, off2;
+	const TfwStr *c1, *c2;
+
+	BUG_ON(hdr->flags & TFW_STR_DUPLICATE);
+	BUG_ON(!name->len);
+
+	if (unlikely(!hdr->len))
+		return -name->len;
+
+	i1 = i2 = 0;
+	off1 = off2 = 0;
+	n = min(hdr->len, name->len);
+	c1 = TFW_STR_CHUNK(hdr, 0);
+	c2 = TFW_STR_CHUNK(name, 0);
+	while (n) {
+		int cn = min(c1->len - off1, c2->len - off2);
+		int r = tfw_cstricmp(c1->data + off1,
+				     c2->data + off2, cn);
```

Comment:
That's not HTTP/2 way: why to compare chunked strings in the loop if we have static table indexes for the most frequent header? Let's compare the indexes and add the indexes for HTTP/1 header strings.